src/asm/section.cpp

```diff
@@ -568,22 +570,43 @@ static void growSection(uint32_t growth) {
 		currentLoadSection->size = curOffset;
 }
 
-static void writebyte(uint8_t byte) {
-	if (uint32_t index = sect_GetOutputOffset(); index < currentSection->data.size())
-		currentSection->data[index] = byte;
-	growSection(1);
+// This should be paired with one or more calls to `growSection` with equivalent `growth`.
+[[gnu::alloc_size(1), gnu::warn_unused_result]]
+static uint8_t *reserveBytes(uint32_t growth) {
+	if (!requireCodeSection())
+		return nullptr;
+
+	// If the section will grow larger than its max size, don't bother writing any bytes to it.
+	// The caller will just give up.
+	if (uint32_t maxSize = sectionTypeInfo[currentSection->type].size;
+	    currentSection->size + growth > maxSize) {
+		// The caller is going to bail and not emit any of the bytes it should.
+		// But, for the reported final section sizes to still be accurate, the section's size must
+		// still be updated. So we do it here, expecting the caller to bail.
+		growSection(growth);
```

Comment:
I would much rather that each caller do the `growSection` itself before bailing, even if that would be repetitive, rather than have this inconsistent interface that's easier to accidentally omit or duplicate a `growSection`.